File: Tmp/Five/feature1.patch.c
diff --git a/eg.c b/eg.c
index e98f05d..c5ff63d 100644
--- a/eg.c
+++ b/eg.c
@@ -5,12 +5,16 @@
 
 int convert(char* name);
 
+void check_strings();
+
 int main(int argc, char** argv){
 
   char name[26] = "abcde\0";
   char greet[26];
   int value;
 
+  check_strings();
+
   strcpy(greet, "Hello");
   printf("%s", greet);
   printf("%c", '\n');
@@ -45,3 +49,19 @@ int convert(char* name){
 
   return converted_val;
 }
+
+void check_strings(){
+
+  char test_string[26];
+  int value;
+
+  strcpy(test_string, "aaaa");
+  if( convert(test_string) != 1) printf("%s", "That's wrong\n");
+
+  strcpy(test_string, "bbbb");
+  if( convert(test_string) == 2) printf("%s", "That's right\n");
+
+  strcpy(test_string, "ddd");
+  if( convert(test_string) == 0) printf("%s", "Not implemented\n");
+
+}

File: Tmp/Five/feature1.patch.f90
diff --git a/eg.f90 b/eg.f90
index d15ab90..385a659 100644
--- a/eg.f90
+++ b/eg.f90
@@ -25,14 +25,41 @@ FUNCTION convert(name)
 END FUNCTION
 END MODULE mod1
 
+MODULE tests
+
+  USE mod1
+IMPLICIT NONE
+  CONTAINS
+
+  SUBROUTINE check_strings
+
+    CHARACTER(26) :: test_string
+    INTEGER :: value
+
+    test_string = "aaaa"
+    IF( convert(test_string) /= 1) PRINT*, "That's wrong"
+
+    test_string = "bbbb"
+    IF( convert(test_string) == 2) PRINT*, "That's right"
+
+    test_string = "ddd"
+    IF( convert(test_string) == 0) PRINT*, "Not implemented"
+
+  END SUBROUTINE
+
+END MODULE tests
+
 PROGRAM test
 
   USE mod1
+  USE tests
   IMPLICIT NONE
 
   CHARACTER(26) :: name, greet
   INTEGER :: value
 
+  CALL check_strings()
+
   greet = "Hello"
   PRINT*, greet
 
